plugins/inputs/ceph_performance/README.md

```diff
@@ -0,0 +1,98 @@
+# Ceph Performance Input Plugin
+
+Collects metricts via ceph commands rather than from the admin socket.  This plugin allows access to metrics for the entire cluster and per pool statistics (e.g. utilization, reads, writes and operations per second).
+
+The following commands are executed and parsed to collect metrics:
+
+* ceph status
+* ceph df
+* ceph osd pool stats
+
+## Configuration
+```
+# Collects I/O and utilization statistics from ceph via the cli.
+[[inputs.ceph_performance]]
+    ## All values are optional and default to those shown
+
+    ## Ceph binary location
+    ceph_binary = "/usr/bin/ceph"
+
+    ## Ceph user to authenticate as
+    ceph_user = "client.admin"
+
+    ## Ceph configuration to use to locate the cluster
+    ceph_config = "/etc/ceph/ceph.conf"
+```
+
+## Measurements & Fields
+
+* ceph.osdmap
+  * epoch (float)
+  * full (boolean)
+  * nearfull (boolean)
+  * num\_in\_osds (float)
+  * num\_osds (float)
+  * num\_remremapped\_pgs (float)
+  * num\_up\_osds (float)
+
+* ceph.pgmap
+  * bytes\_avail (float)
+  * bytes\_total (float)
+  * bytes\_used (float)
+  * data\_bytes (float)
+  * num\_pgs (float)
+  * op\_per\_sec (float)
+  * read\_bytes\_sec (float)
+  * version (float)
+  * write\_bytes\_sec (float)
+  * recovering\_bytes\_per\_sec (float)
+  * recovering\_keys\_per\_sec (float)
+  * recovering\_objects\_per\_sec (float)
+
+* ceph.pgmap.state
+  * value (float)
+
+* ceph.usage
+  * bytes\_used (float)
+  * kb\_used (float)
+  * max\_avail (float)
+  * objects (float)
+
+* ceph.pool.usage
+  * bytes\_used (float)
+  * kb\_used (float)
+  * max\_avail (float)
+  * objects (float)
+
+* ceph.pool.stats
+  * op\_per\_sec (float)
+  * read\_bytes\_sec (float)
+  * write\_bytes\_sec (float)
+  * recovering\_object\_per\_sec (float)
+  * recovering\_bytes\_per\_sec (float)
+  * recovering\_keys\_per\_sec (float)
+
+## Tags:
+* ceph.pg.state has the following tags:
+  * state (state for which the value applies e.g. active+clean, active+remapped+backfill)
+* ceph.pool.usage has the following tags:
+  * id
+  * name
+* ceph.pool.stats has the following tags:
+  * id
+  * name
+
+## Example Output
+<pre>
+$ telegraf -config telegraf/telegraf.conf -test -input-filter ceph_performance
+* Plugin: ceph_performance, Collection 1
+* Internal: 1m0s
+> ceph.osdmap,host=ceph-mon-0 epoch=170772,full=false,nearfull=false,num_in_osds=340,num_osds=340,num_remapped_pgs=0,num_up_osds=340 1468841037000000000
+> ceph.pgmap,host=ceph-mon-0 bytes_avail=634895531270144,bytes_total=812117151809536,bytes_used=177221620539392,data_bytes=56979991615058,num_pgs=22952,op_per_sec=15869,read_bytes_sec=43956026,version=39387592,write_bytes_sec=165344818 1468841037000000000
+> ceph.pgmap.state,host=ceph-mon-0,state=active+clean value=22952 1468841037000000000
```

Comment:
This is an interesting one, the states only show when they have a count.  So trying to graph this in Grafana is next to impossible unless all possible fields have been seen.  The general idea was a query like: `SELECT value FROM ceph_pgmap_state GROUP BY state`, unless there is a different way of achieving this...